htlcswitch/hop/iterator.go

```diff
@@ -113,6 +120,98 @@ func (r *sphinxHopIterator) ExtractErrorEncrypter(
 	return extracter(r.ogPacket.EphemeralKey)
 }
 
+// BlindingProcessor is an interface that provides the cryptographic operations
+// required for processing blinded hops.
+type BlindingProcessor interface {
+	// DecryptBlindedHopData decrypts a blinded blob of data using the
+	// ephemeral key provided.
+	DecryptBlindedHopData(ephemPub *btcec.PublicKey,
+		encryptedData []byte) ([]byte, error)
+}
+
+// BlindingKit contains the components required to extract forwarding
+// information for hops in a blinded route.
+type BlindingKit struct {
+	// BlindingPoint holds a blinding point that was passed to the node via
+	// update_add_htlc's TLVs.
+	BlindingPoint *btcec.PublicKey
```

Comment:
IIUC, this is always provided, so we could get away with just including the raw value and not a pointer. I think at times in the code base due to historical pattern matching, we end up passing around more pointers to stuff than we actually need to (no need to mutate, not worried about copying some massive thing). 

I think this is where (even with some of the akwardness) wrappers like `fn.Option` are useful, as they let you distinguish between something that may be there, vs a pointer that the function you called returned, so now you also carry along that pointer.

Reply:
This won't be provided in the case where we're the introduction node and the `BlindingPoint` is in the `Payload`. Updated to have a non-pointer `BlindingKit` because we always need it _in case_ we're the introduction node and used `Option` record for blinding point instead.